docker-app/qfieldcloud/core/views/files_views.py

```diff
@@ -135,25 +139,54 @@ def get(self, request, projectid, filename):
 
     # TODO refactor this function by moving the actual upload and Project model updates to library function outside the view
     def post(self, request, projectid, filename, format=None):
-        project = Project.objects.get(id=projectid)
+        """Handle POST requests."""
+
+        # FIXME: CURRENTLY UNDERGOING DEBUG
+        # Write traceback to buffer + deepcopy request before anything happens to the request object
+        buffer = io.StringIO()
+        print_stack(limit=50, file=buffer)
+
+        # Fetch important data without mutating request
+        request_attributes = {
+            "data": str(copy.copy(request.data).keys()),
+            "files": str(request.FILES.keys()),
+            "meta": str(request.META),
+        }
+
+        if len(request.FILES.getlist("file")) > 1:
+            raise exceptions.MultipleContentsError()
+
+        missing_error = ""
 
         if "file" not in request.data:
-            logger.info(
-                'The key "file" was not found in `request.data`.',
-                extra={
-                    "data_for_key_text_content": str(request.data.get("text", ""))[
-                        :1000
-                    ],
-                    "data_for_key_text_len": len(request.data.get("text", "")),
-                    "request_content_length": request.META.get("CONTENT_LENGTH"),
-                    "request_content_type": request.META.get("CONTENT_TYPE"),
-                    "request_data": list(request.data.keys()),
-                    "request_files": list(request.FILES.keys()),
-                },
-            )
+            missing_error = 'The key "file" was not found in `request.data`. Sending report to Sentry.'
+
+        if not request.FILES.getlist("file"):
+            missing_error = 'The key "file" occurs in `request.data` but maps to an empty list. Sending report to Sentry.'
+
+        # When missing_error is truthy...
+        if missing_error:
+            logging.warning(missing_error)
+
+            # ...report to sentry if the length is below the 10MBs
+            if int(request.META.get("CONTENT_LENGTH")) < 10000000:
+                report_to_sentry(
+                    # using the 'X-Request-Id' added to the request by RequestIDMiddleware
+                    name=f"{request.META.get('X-Request-Id')}_{projectid}",
+                    request_attributes=request_attributes,
+                    # no need to read the buffer just yet
+                    buffer=buffer,
+                )
             raise exceptions.EmptyContentError()
 
+        # get project from request or db
+        project = (
+            request.project
+            if hasattr(request, "project")
+            else Project.objects.get(id=projectid)
+        )
```

Comment:
```suggestion
        if hasattr(request, "project"):
            project = request.project
        else:
            project = Project.objects.get(id=projectid)
```

Fewer lines, much easier for lazy brains like mine to follow.
Avoid ternary logic (except for ... I'm sure there's some exception)

Reply:
Actually, that's a change to before (where it was always taken from the DB), is there a reason?

Reply:
It should be reverted to what it was. I tried a suggestion from Ivan, and the suggestion was sound.

Reply:
> ```suggestion
>         if hasattr(request, "project"):
>             project = request.project
>         else:
>             project = Project.objects.get(id=projectid)
> ```
> 
> Fewer lines, much easier for lazy brains like mine to follow.
> Avoid ternary logic (except for ... I'm sure there's some exception)

I disagree that it's more effort, since it makes immediately clear that the variable cannot be unbound and cannot be unsassigned, however I am happy to accommodate the opposite view.

Reply:
Please change

Reply:
I would suggest we revert this to the original `project = Project.objects.get(id=projectid)`, but it is important to know what situations the `request.project` is not working... In my understanding it should work in any case and if it doesn't, there is a deeper issue we can investigate in another PR.